base/src/jvmMain/kotlin/utils/Date.kt

```diff
@@ -0,0 +1,89 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base.utils
+
+import java.util.Calendar
+
+actual class Date(internal val calendar: Calendar) : Comparable<Date> {
+
+    actual companion object {
+        actual fun now(offsetInMilliseconds: Long, timeZone: TimeZone, locale: Locale): Date = Date(Calendar.getInstance(timeZone.timeZone, locale.locale).apply {
+                add(Calendar.MILLISECOND, offsetInMilliseconds.toInt())
+            })
+        actual fun epoch(offsetInMilliseconds: Long, timeZone: TimeZone, locale: Locale): Date = Date(Calendar.getInstance(timeZone.timeZone, locale.locale).apply {
+            timeInMillis = offsetInMilliseconds
+        })
+    }
+
+    actual var timeZone: TimeZone
+        get() = TimeZone(calendar.timeZone)
+        set(value) { calendar.timeZone = value.timeZone }
+
+    actual var era: Int
+        get() = calendar.get(Calendar.ERA)
+        set(value) { calendar.set(Calendar.ERA, value) }
+    actual var year: Int
+        get() = calendar.get(Calendar.YEAR)
+        set(value) { calendar.set(Calendar.YEAR, value) }
+    actual var month: Int
+        get() = calendar.get(Calendar.MONTH) + 1
+        set(value) { calendar.set(Calendar.MONTH, value - 1) }
+    actual var weekOfYear: Int
+        get() = calendar.get(Calendar.WEEK_OF_YEAR)
+        set(value) { calendar.set(Calendar.WEEK_OF_YEAR, value) }
+    actual var weekOfMonth: Int
+        get() = calendar.get(Calendar.WEEK_OF_MONTH)
+        set(value) { calendar.set(Calendar.WEEK_OF_MONTH, value) }
+    actual var day: Int
+        get() = calendar.get(Calendar.DAY_OF_MONTH)
+        set(value) { calendar.set(Calendar.DAY_OF_MONTH, value) }
+    actual var dayOfYear: Int
+        get() = calendar.get(Calendar.DAY_OF_YEAR)
+        set(value) { calendar.set(Calendar.DAY_OF_YEAR, value) }
+    actual var weekDay: Int
+        get() = calendar.get(Calendar.DAY_OF_WEEK) + 1
+        set(value) { calendar.set(Calendar.DAY_OF_WEEK, value - 1) }
+
+    actual var hour: Int
+        get() = calendar.get(Calendar.HOUR_OF_DAY)
+        set(value) { calendar.set(Calendar.HOUR_OF_DAY, value) }
+    actual var minute: Int
+        get() = calendar.get(Calendar.MINUTE)
+        set(value) { calendar.set(Calendar.MINUTE, value) }
+    actual var second: Int
+        get() = calendar.get(Calendar.SECOND)
+        set(value) { calendar.set(Calendar.SECOND, value) }
+    actual var millisecond: Int
+        get() = calendar.get(Calendar.MILLISECOND)
+        set(value) { calendar.set(Calendar.MILLISECOND, value) }
+    actual var millisecondSinceEpoch: Long
+        get() = calendar.timeInMillis
+        set(value) { calendar.timeInMillis = value }
+
+    actual fun copy(): Date = Date(calendar.clone() as Calendar)
+
+    override fun equals(other: Any?): Boolean {
+        return (other as? Date)?.let {
+            calendar == it.calendar
```

Comment:
I would throw an exception if types cannot be compared

Reply:
Thats not what kotlin usually does when overwriting equals AFAIK. So I wouldnt be in favor of that personally